Modules/_ctypes/callproc.c

```diff
@@ -1893,22 +1899,35 @@ unpickle(PyObject *self, PyObject *args)
     return NULL;
 }
 
+/*[clinic input]
+_ctypes.POINTER as create_pointer_type
+
+    type: object
+    /
+
+Creates and returns a new ctypes pointer type.
+
+Pointer types are cached and reused internally, so calling this function
+repeatedly is cheap. 'type' must be a ctypes type.
+[clinic start generated code]*/
+
 static PyObject *
-POINTER(PyObject *self, PyObject *cls)
+create_pointer_type(PyObject *module, PyObject *type)
+/*[clinic end generated code: output=6f3fdd0cf5308ce6 input=fcc2effb05dc7ab5]*/
 {
     PyObject *result;
     PyTypeObject *typ;
     PyObject *key;
 
-    result = PyDict_GetItemWithError(_ctypes_ptrtype_cache, cls);
+    result = PyDict_GetItemWithError(_ctypes_ptrtype_cache, type);
```

Comment:
The docs call the argument `type` so I renamed `cls` to `type` here, but then there's another local variable called `typ` which is a bit confusing :confused:

Reply:
You can preserve the old argument name in C code by defining `type` argument inside clinic input block as `type as cls: object`.

Reply:
Thanks for the tip!